tutorial/Tutorial.md

```diff
@@ -0,0 +1,137 @@
+
+
+## Setup and Code Example 
+
+Initial requirements:
+- [Install smove](https://github.com/eigerco/smove)
+- Setup a substrate node with pallet-move integrated or use [our template node](https://github.com/eigerco/substrate-node-template-move-vm-test)
+- Switch current working directory to the code example directory in `pallet-move/tests/assets/move-projects/car-wash-example`
+
+In our example, a simple car wash is modelized, where washing coins are used to control the usage of the car wash. Users can register, buy washing coins and use them to start the car wash. For the sake of simplicity, we show here only the function headers of the Move module, the full module can be seen [here](https://github.com/eigerco/pallet-move/blob/main/tests/assets/move-projects/car-wash-example/sources/CarWash.move).
+
+```move
+module DeveloperBob::CarWash {
+    /// Struct stores number of coins for each user.
+    struct Balance has key, store {
+        coins: u8
+    }
+
+    /// Method executes the ICO without money. The module owner (also the car wash owner) gets deposited the minted washing coins.
+    public fun initial_coin_minting(module_owner: &signer) {}
+
+    /// Registers a new user. The account address will be added to the storage Balance with zero initial washing coins.
+    public fun register_new_user(account: &signer) {}
+
+    /// Buys a washing coin for the car wash. Therfore, `COIN_PRICE` will be withdrawn from the user's account.
+    // Note: It would be nice to have the ability to buy multiple coins at once, but it's not implemented for this example.
+    public fun buy_coin(user: &signer) acquires Balance {}
+
+    /// Initiates the washing process by paying one washing coin.
+    public fun wash_car(user: &signer) acquires Balance {}
+}
+```
+
+Each publically usable method will be depicted into an executable script, like this style:
+```move
+script {
+    use DeveloperBob::CarWash;
+    
+    fun initial_coin_minting(account: signer) {
+        CarWash::initial_coin_minting(&account);
+    }
+}
+```
+
+Now let's compile this project to be ready for estimation of needed gas and publication of the module.
+```sh
+smove build
+```
+
+## Gas Estimation for Publishing a Module
+
+Gas can be seen as weights in the Substrate. It will adjust the fees for the execution or publication of Move resources. Hereby, we prevent malicious scripts from running forever.
+
+To estimate the optimal amount of gas for your module, use `smove` tool:
+```sh
+smove node rpc estimate-gas-publish-module --account-id 5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty --module-path build/car-wash-example/bytecode_modules/CarWash.mv
+```
+
+The successful result will look like:
+```sh
+Estimated gas: Estimate (gas_used: 69, vm_status_code: EXECUTED)
+```
+
+## Publishing Modules
+
+The compiled bytecode file can be found in the subfolder
+```sh
+build/car-wash-example/bytecode_modules
+```
+
+Do the following steps in polkadot.js:
```

Comment:
Maybe we can insert a link here (https://polkadot.js.org/apps)
```suggestion
Do the following steps in polkadot.js GUI:
```

Reply:
Done